music21/musicxml/m21ToXml.py

```diff
@@ -3322,8 +3370,26 @@ def parseOneElement(self, obj):
         self.objectSpannerBundle = self.spannerBundle.getBySpannedElement(obj)
         preList, postList = self.prePostObjectSpanners(obj)
 
-        for sp in preList:  # directions that precede the element
-            root.append(sp)
+        if prePostObjectSpannersOnly:
+            if not preList and not postList:
```

Comment:
I think I'd like to see this as three stages -- 

(1) write the preList elements unless AppendSpanners.NONE.  
(2) write the object...unless AppendSpanners.RELATED_ONLY in which case move forward.
(3) write the postList elements unless AppendSpanners.NONE

The logic here is now much more confusing to me and requires updating multiple places for any later developments.

Reply:
Yes!  Much better.